lib/versioning/swift/index.ts

```diff
@@ -37,7 +37,7 @@ const getSatisfyingVersion = (versions: string[], range: string): string =>
   );
 const minSatisfyingVersion = (versions: string[], range: string): string =>
   minSatisfying(
-    versions.map((v) => v.replace(regEx(/^v/), '')), // TODO #12071 #12070
+    versions.map((v) => v.replace(regEx(/^v/), '')), // TODO #12070
```

Comment:
Are you sure about the todo?

Reply:
It's inside a map function so it will have to be created each time.

I have a doubt regarding this which is like this:-
1) I create a regex like this - `re = /abs/;`
2) I create a regex like this - `re = new re2(/abs/);`

in 1)  from the Regex docs we know that even if it is inside a loop  it will only be constructed once.
But I couldn't fond any info about the creation of regex in 2) case which leaves me with doubt when I encounter such regex

If we use a string to create regex then it will be re-constructed for each iteration. For now I think it is going to be same in cases where we use an object to create a regex.

Reply:
It's inside a map function so it will have to be created each time.

I have a doubt regarding this which is like this:-

I create a regex like this - re = /abs/;
I create a regex like this - re = new re2(/abs/);
in 1) from the Regex docs we know that even if it is inside a loop it will only be constructed once.
But I couldn't fond any info about the creation of regex in 2) case which leaves me with doubt when I encounter such regex

If we use a string to create regex then it will be re-constructed for each iteration. For now I think it is going to be same in cases where we use an object to create a regex.

Reply:
As told above `regEx` function is caching all re2 instances, so you can call that function with same string / regex multiple times and always get same re2 instance back. So the only overhead is the function call and a cache lookup.